src/wutil.cpp

```diff
@@ -311,7 +311,7 @@ int fd_check_is_remote(int fd) {
             // Other FSes are assumed local.
             return 0;
     }
-#elif defined(MNT_LOCAL)
+#elif defined(MNT_LOCAL) && !defined(__NetBSD__)
```

Comment:
On NetBSD you must use POSIX interface statvfs(2) in the place of statfs().

Reply:
I know. We might switch to that later (though we require a MNT_LOCAL-like flag, and my fstatvfs doesn't have ST_LOCAL, so this might be a bit more tricky than just switching everywhere). But as it stands, all this does is to cause fish to not use mmap on the history file.

So it's not really all that important.

Reply:
Do you mean that you don't use nfs so you disable feature for everybody? Adding support for it is mechanical as the code is similar to code here, but on vfs struct rather than fs.

Please fix it.

Reply:
https://github.com/llvm-mirror/llvm/blob/802db564f99e0fb93cbe69f7c05b48f0400f559d/lib/Support/Unix/Path.inc#L372 Similar code in LLVM.

Reply:
We have other strategies of dealing with the history. It's a relatively minor optimization.

Reply:
I don't want to see NetBSD as leftover. If you reject doing it, I will submit a patch for it myself.

Reply:
I'm not outright rejecting it, I'm saying we can do it later, properly, after investigating on which systems we can use which code.

NetBSD now _builds and runs_, even on wscon (something your package doesn't do!) so it's already much less of a leftover. The fact that using the history file takes a few microseconds longer isn't much of a concern, in comparison.

Reply:
Okay, it turns out the only system that allows us to use the POSIX interface here is NetBSD, because _all_ other systems added the non-standard extension to the non-standard function, while NetBSD extended the standard function with the non-standard extension.

@krytarowski: See 25983dd.

Reply:
> I don't want to see NetBSD as leftover.

That's really not fair - @faho is not a BSD guy, but spent a lot of time installing NetBSD and getting this going for the sake of making sure fish works well on an OS he doesn't use personally. And he's spent a lot of time these last few days trying to get continuous integration going for NetBSD and FreeBSD, so in the future we never accidentally break things on these platforms. NetBSD is very much getting the first-class treatment here.

Reply:
I do appreciate the effort and ensure to keep the feedback from the NetBSD community. I wrote an original set of patches in pkgsrc-wip for a modern fish-shell, they got mutated and are landing here.

Thanks!